### PR TITLE
[81] Provide a way to access to all tools contained in diagram inner Toolbar when Diagram size is reduced

### PR DESCRIPTION
**Pull request by scosta-obeo**

### CODE REVIEW

**Review by florianbarbin — COMMENTED** [review]
We probably should migrate this issue https://bugs.eclipse.org/bugs/show_bug.cgi?id=582149 on github

File: plugins/org.eclipse.sirius.tests.swtbot/src/org/eclipse/sirius/tests/swtbot/tabbar/TabBarTest.java [hunk 1]
```diff
@@ -534,4 +537,58 @@ private void checkDiagramTabbarButtons(boolean activeExtensions) {
         }
         checkButtonNotPresent(TABBAR_EXTENSION_ON_DIAGRAM_ELEMENT);
     }
+
+    public void testToolBarWrap() throws Exception {
```
Comment (florianbarbin) [inline plugins/org.eclipse.sirius.tests.swtbot/src/org/eclipse/sirius/tests/swtbot/tabbar/TabBarTest.java:541]: Just add a comment to explain what is verified by this test

File: plugins/org.eclipse.sirius.tests.swtbot/src/org/eclipse/sirius/tests/swtbot/tabbar/TabBarTest.java [hunk 2]
```diff
@@ -534,4 +537,58 @@ private void checkDiagramTabbarButtons(boolean activeExtensions) {
...
+        }
+
+        // we check that no tools are outside the visible part of the toolbar
+        for (SWTBotToolbarButton item : toolbarDiagramItems) {
+            boolean itemIsOutside = syncExec(() -> {
+                int parentWidth = item.widget.getParent().getSize().x;
+                int rightPos = item.widget.getBounds().x + item.widget.getBounds().width;
+                return rightPos <= parentWidth;
+            });
+            assertTrue("ToolButton `" + item.getToolTipText() + "` is outside the visible part of the toolbar.", itemIsOutside);
```
Comment (florianbarbin) [inline plugins/org.eclipse.sirius.tests.swtbot/src/org/eclipse/sirius/tests/swtbot/tabbar/TabBarTest.java:566]: The test fail on my environment: junit.framework.AssertionFailedError: ToolButton `Copy the format of the whole diagram` is outside the visible part of the toolbar.

File: plugins/org.eclipse.sirius.tests.swtbot/src/org/eclipse/sirius/tests/swtbot/tabbar/TabBarTest.java [hunk 3]
```diff
@@ -534,4 +537,58 @@ private void checkDiagramTabbarButtons(boolean activeExtensions) {
...
+        for (int i = 0; i < DIAGRAM_TOOLBARBUTTONS_TOOLTIPS.length; ++i) {
+            toolbarDiagramItems.add(editor.bot().toolbarButton(i));
+        }
+        for (int i = 0; i < DIAGRAM_TOOLBARDROPDOWNBUTTONS_TOOLTIPS.length; ++i) {
+            toolbarDiagramItems.add(editor.bot().toolbarDropDownButton(i));
+        }
+
+        // we check that no tools are outside the visible part of the toolbar
+        for (SWTBotToolbarButton item : toolbarDiagramItems) {
+            boolean itemIsOutside = syncExec(() -> {
```
Comment (florianbarbin) [inline plugins/org.eclipse.sirius.tests.swtbot/src/org/eclipse/sirius/tests/swtbot/tabbar/TabBarTest.java:561]: itemIsOutside => you mean itemIsInside ?

**Review by florianbarbin — DISMISSED** [review]

**Review by lfasani — APPROVED** [review]

**Pull request merged**